_posts/en/newsletters/2019-04-02-newsletter.md

```diff
@@ -0,0 +1,269 @@
+  more in fees than if she chose the route herself.
+
+- **Bitcoin Core schedules switch to default bech32 receiving
+  addresses:** since [version 0.16.0][0.16.0 segwit], Bitcoin Core's
+  built-in wallet has defaulted to generating P2SH-wrapped segwit
+  addresses when users want to receive payments.  These addresses are
+  backwards compatible with all widely-used software.  As
+  discussed in an [issue][Bitcoin Core #15560] and [the project's weekly
+  meeting][core meeting segwit], starting with Bitcoin Core 0.20
+  (expected about a year from now), Bitcoin Core will default to native
+  segwit addresses (bech32) that provide additional fee savings and
+  other benefits.  Currently, many wallets and services [already support
+  bech32 addresses][bech32 adoption], and if the Bitcoin Core project
+  sees enough additional adoption in the next six months to warrant an
+  earlier switch, it will instead default to bech32 receiving addresses
+  in Bitcoin Core 0.19.  Legacy address will continue to be provided
+  if the user requests them in the GUI or by RPC, and anyone who doesn't
+  want the update will be able to configure their default address type.
+  (Similarly, pioneering users who want to change their default now may
+  set the `addresstype=bech32` configuration option in any Bitcoin Core
+  release from 0.16.0 up.)
+
+## Bech32 sending support
+
+*Week 3 of 24.  From now until the second anniversary of the segwit soft
+fork lock-in on 24 August 2019, the Optech Newsletter will contain this
+weekly section that provides information to help developers and
+organizations implement bech32 sending support---the ability to pay
+native segwit addresses.  This [doesn't require implementing
+segwit][bech32 easy] yourself, but it does allow the people you pay to
+access all of segwit's multiple benefits.*
+
+In a [previous week][bech32 easy], we discussed how small the
+differences are between creating the output for a legacy address versus
+a native segwit address.  In that section we simply pointed you towards
+the [bech32 reference libraries][] and told you that you'd get two
+values back.  In this week, we walkthrough the exact steps of using the
+Python reference library so you can see how little work this is.  We
+start by importing the library:
+
+```python3
+import segwit_addr
+```
+
+Bech32 addresses have a Human-Readable Part (HRP) that indicates what
+network the address is for.  These are the first few characters of the
+address and are separated from the data part of the address by the
+delimiter `1`.  For example, Bitcoin testnet uses `tb` and an example
+testnet address is tb1q3w[...]g7a.  We'll set the Bitcoin mainnet HRP of
+`bc` in our code so that we can later ensure that the addresses we parse
+are for the network we expect.
+
+```python3
+HRP='bc'
+```
+
+Finally, we have a few addresses we want to check---one that should work
+and two that should fail.  (See [BIP173][] for a complete set of
+reference test vectors.)
```

Comment:
optional micro-optimization: deep link to the test vector section? https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki#Test_vectors